wagtail_extensions/models.py

```diff
@@ -64,9 +71,13 @@ def primary_opening_times(self):
 
     @property
     def primary_opening_today(self):
-        opening_times = self.primary_opening_times
-        if opening_times:
-            today = date.today()
-            specific_times = utils.first_true(opening_times, lambda x: x.get('date') == today)
-            return specific_times or utils.first_true(opening_times, lambda x: x.get('weekday') == today.weekday())
-        return None
+        today = date.today()
+        cache_key = self.get_opening_today_cache_key(today)
+        times = cache.get(cache_key)
+        if not times:
```

Comment:
Better to check `if times is not None`, otherwise you will repeatedly miss if opening times list happens to be empty.